packages/waas/src/auth.ts

```diff
@@ -383,7 +383,13 @@ export class SequenceWaaS {
     return this.waas.isSignedIn()
   }
 
-  signIn(creds: Identity, sessionName: string): Promise<SignInResponse> {
+  async signIn(creds: Identity, sessionName: string): Promise<SignInResponse> {
+    // We clear and drop session regardless of whether it's signed in or not
+    const currentSessionId = await this.waas.getSessionId()
+    if (currentSessionId) {
+      await this.dropSession({ sessionId: currentSessionId, strict: false })
```

Comment:
does .dropSession make a call to the waas service..? what if this raises an exception / fails..?

Reply:
Im just mentioning so we ensure it fails gracefully etc

Reply:
`strict: false` allows it to do it gracefully and continue to clear stuff regardless

```ts
  async dropSession({ sessionId, strict }: { sessionId?: string; strict?: boolean } = {}) {
    const thisSessionId = await this.waas.getSessionId()
    if (!thisSessionId) {
      throw new Error('session not open')
    }

    const closeSessionId = sessionId || thisSessionId

    try {
      const intent = await this.waas.signOutSession(closeSessionId)
      const result = await this.sendIntent(intent)

      if (!isCloseSessionResponse(result)) {
        throw new Error(`Invalid response: ${JSON.stringify(result)}`)
      }
    } catch (e) {
      if (strict) {
        throw e
      }

      console.error(e)
    }

    if (closeSessionId === thisSessionId) {
      if (!this.secureStoreBackend) {
        throw new Error('No secure store available')
      }

      const session = await newSessionFromSessionId(thisSessionId, this.cryptoBackend, this.secureStoreBackend)
      session.clear()
      await this.waas.completeSignOut()
      await this.deviceName.set(undefined)
    }
  }
  ```